soh/soh/Enhancements/randomizer/draw.cpp

```diff
@@ -0,0 +1,116 @@
+#include "draw.h"
+#include "z64.h"
+#include "macros.h"
+#include "functions.h"
+#include "randomizerTypes.h"
+#include <array>
+#include "objects/object_gi_key/object_gi_key.h"
+#include "objects/object_gi_bosskey/object_gi_bosskey.h"
+#include "objects/object_gi_hearts/object_gi_hearts.h"
+
+extern "C" void Randomizer_DrawSmallKey(GlobalContext* globalCtx, GetItemEntry* getItemEntry) {
```

Comment:
How does this currently handle Thieve's Hideout small keys? I'm not really seeing an exception for them. Or are they not using this function at all and still using the "vanilla" one?

Reply:
Correct, they just don't have a draw function in their entry and just use the vanilla one.

Reply:
I'm not entirely sure how to feel about this. On one hand, it feels weird making that key use this function and then exclude it from applying color in the function itself, on the other hand it feels weird having it be the only key not using the function, especially considering the name of it.

Reply:
The way I architected the drawing functions it makes more sense to have it not use this function at all. Take a look in `randomizer.cpp` where I'm creating the item table. While I'm looping through the array of item entries I defined, I add this drawing function to the keys that need it. Thankfully RG_GERUDO_FORTRESS_SMALL_KEY was the first one in the list. so I didn't have to special-case it, could just apply to everything from Forest Temple to Ganon's Castle.

Reply:
Actually wait, I'm wrong on this one. The keys would be colored incorrectly at the moment. Not sure what I saw then. I'll have to add an exception but I know where I can add it that I shouldn't need to change this function much.

Reply:
Ok this should be fixed now.